restful.module

```diff
@@ -527,6 +527,9 @@ function restful_menu_process_callback($resource_name, $version = NULL) {
     foreach ($e->getHeaders() as $header_name => $header_value) {
       drupal_add_http_header($header_name, $header_value);
     }
+
+    $severity = $e->getCode() < 500 ? WATCHDOG_NOTICE : WATCHDOG_ERROR;
+    watchdog('restful', $e->getMessage(), array(), $severity);
```

Comment:
Didn't know about `watchdog_exception ` :) I think we should use it only for the below case and when status code >= 500.